### ISSUE TITLE
About NuScenes

### ISSUE THREAD

**Issue by PSYZ1234** [ispc-lab/LiDAR4D#6]
Hi, thank you for your good paper with beautiful code. I want to know how to train and test (data split) on the NuScenes dataset, and would you consider releasing the corresponding code?

**Comment by dyfcalid** [comment]
We have not yet organized the relevant code for NuScenes, but you can refer to the KITTI-360 dataset code for modification.  And we use [NuScenes-Mini](https://www.nuscenes.org/nuscenes#download) for convenience. The scenario sequence is shown in Appendix Table S6, which represents the index number of the v1.0-mini dataset (merged `LIDAR_TOP` in the `samples` and `sweeps` folders). The data split is the same as KITTI-360 dynamic sequences.

**Comment by PSYZ1234** [comment]
OK, thank you

**Comment by PSYZ1234** [comment]
Hi, I have another question. For the pose loading of nuscenes, should I load the corresponding pose in ego_pose.json? Do I need to make any transformation, such as applying the translation and rotation of calibrated_sensor.json or other files?

**Comment by dyfcalid** [comment]
Hi.
Right, you can refer to the official utils in nuscenes to obtain the sensor poses. Specifically, you can get the transform_matrix `lidar2car` from `calibrated_sensor` and `car2world` from `ego_pose`. Then the final pose `lidar2world` would be `car2world@lidar2car`.

**Comment by PSYZ1234** [comment]
Thank you for your reply. I have obtained the correct pose. 
However, during the training process, I found that except for the Seq 450-500, the loss of all other sequences became NAN after training for a few steps in the first epoch. I haven't changed any parameter settings. Is there anything wrong with my experimental setup?

**Comment by dyfcalid** [comment]
Hello.
Maybe you should visualize the point cloud sequence first to see if it is correct and continuous.
After that, check the configuration of the nuscenes dataset.
For example, modify the `H` `W` `intrinsics` for range-view generation, and calculate `scale` `offset`  of this sequence to generate corresponding `txt` config file. 
The following is recommended: `H = 32` `W = 1024` `fov_lidar = [10.0, 40.0]`.

**Comment by PSYZ1234** [comment]
Thank you very much for your answer. I will continue to try and if there are any questions, I will ask you again.

**Comment by PSYZ1234** [comment]
Hi, sorry to bother you.
I want to confirm with you that the Seq ID I have chosen is correct. Based on the sampling frequency of nuScenes (20HZ) given in the paper, which is consistent with KITTI (10HZ), I selected one frame per frame interval in the sequences. 

The corresponding starting and ending Seq IDs are as follows:
Seq 450-500: n008-2018-08-01-15-16-36-0400__LIDAR_TOP__1533151606948845.pcd -> n008-2018-08-01-15-16-36-0400__LIDAR_TOP__1533151611946499.pcd
Seq 1250-1300: n008-2018-08-28-16-43-51-0400__LIDAR_TOP__1535489300196771.pcd -> n008-2018-08-28-16-43-51-0400__LIDAR_TOP__1535489305246972.pcd
Seq 1600-1650: n008-2018-08-30-15-16-55-0400__LIDAR_TOP__1535657110100754.pcd -> n008-2018-08-30-15-16-55-0400__LIDAR_TOP__1535657115097816.pcd
Seq 2200-2250: n015-2018-10-02-10-50-40+0800__LIDAR_TOP__1538448756347324.pcd -> n015-2018-10-02-10-50-40+0800__LIDAR_TOP__1538448761348305.pcd
Seq 3180-3230: n015-2018-11-21-19-38-26+0800__LIDAR_TOP__1542800849297337.pcd -> n015-2018-11-21-19-38-26+0800__LIDAR_TOP__1542800854297767.pcd

According to my visualization, the overall shape is similar to that of the paper. But I'm not sure if the frame ID and sampling strategy I've chosen are correct. Thank you.

**Comment by dyfcalid** [comment]
Hi, there.
It looks right and you can just have a try. 🙌

**Comment by PSYZ1234** [comment]
Thank you. I have successfully run the code and obtained the results. 
At present, the results of the first four trajectories are all good, but the prediction error of the last trajectory (Seq 3180-3230) is very large (CD is more than 1). I think this may be due to my inaccurate training and testing splits. 
Does the test frame add 10 to the training start ID, such as 3190, 3200, 3210, and 3220 for Seq 3180-3230? However, the error in this result is significant. Is 3189 or 3191 used for testing (CD is ~0.4)?

**Comment by dyfcalid** [comment]
I think the error may not be due to the test frames. In my experiments both are fine and the results are similar. The CD results of Seq 3180 are around 0.2-0.3 but I have no idea why it only fails for this sequence in your case.

**Comment by PSYZ1234** [comment]
I have resolved this issue, and it is indeed due to my incorrect training and test splits. The default starting frame for KITTI360 is frame 0, while nuScenes has its unique naming convention, so the starting frame should be frame 1. Now I have obtained the same point cloud as the visualization provided in the paper and the results are close to those in the table. 
Thank you so much for your help. I will close this issue.

**Comment by h8c2** [comment]
@PSYZ1234, Hello, I'm facing some challenges with the nuScenes dataset as well. If it's not too much trouble, could you share your data processing scripts?

**Comment by PSYZ1234** [comment]
> @PSYZ1234, Hello, I'm facing some challenges with the nuScenes dataset as well. If it's not too much trouble, could you share your data processing scripts?

Sure, please provide your email, and I will send it to you within the next few days.

**Comment by h8c2** [comment]
> > @PSYZ1234, Hello, I'm facing some challenges with the nuScenes dataset as well. If it's not too much trouble, could you share your data processing scripts?
> 
> Sure, please provide your email, and I will send it to you within the next few days.



> > @PSYZ1234, Hello, I'm facing some challenges with the nuScenes dataset as well. If it's not too much trouble, could you share your data processing scripts?
> 
> Sure, please provide your email, and I will send it to you within the next few days.

Thank you so much for your response; it’s really helpful to me. My email is houchao1999@gmail.com.

**Comment by h8c2** [comment]
@PSYZ1234,  Hello, I hope this message finds you well. I am writing to follow up on the email you mentioned sending a few days ago. I haven’t received it yet and wanted to ensure my email address (houchao1999@gmail.com) was recorded correctly. Please let me know if there's any update or if the email has already been sent. Thanks.

**Comment by PSYZ1234** [comment]
> @PSYZ1234, Hello, I hope this message finds you well. I am writing to follow up on the email you mentioned sending a few days ago. I haven’t received it yet and wanted to ensure my email address ([houchao1999@gmail.com](mailto:houchao1999@gmail.com)) was recorded correctly. Please let me know if there's any update or if the email has already been sent. Thanks.

I sent it again through a new email, have you received it?

**Comment by h8c2** [comment]
> > @PSYZ1234, Hello, I hope this message finds you well. I am writing to follow up on the email you mentioned sending a few days ago. I haven’t received it yet and wanted to ensure my email address ([houchao1999@gmail.com](mailto:houchao1999@gmail.com)) was recorded correctly. Please let me know if there's any update or if the email has already been sent. Thanks.
> 
> I sent it again through a new email, have you received it?

Many thanks to you. I have received the email.

**Comment by booker-max** [comment]
> > @PSYZ1234, Hello, I'm facing some challenges with the nuScenes dataset as well. If it's not too much trouble, could you share your data processing scripts?
> 
> Sure, please provide your email, and I will send it to you within the next few days.



> > @PSYZ1234, Hello, I'm facing some challenges with the nuScenes dataset as well. If it's not too much trouble, could you share your data processing scripts?
> 
> Sure, please provide your email, and I will send it to you within the next few days.

Hello, my email is jin01wang@mail.ustc.edu.cn. Can you provide me with a copy of the preprocess script of nuscenes dataset? Thank you very much.

**Comment by jinxxo-j** [comment]
> > [@PSYZ1234](https://github.com/PSYZ1234), Hello, I'm facing some challenges with the nuScenes dataset as well. If it's not too much trouble, could you share your data processing scripts?
> 
> Sure, please provide your email, and I will send it to you within the next few days.


Hello, my email is [0126jang@yonsei.ac.kr](mailto:0126jang@yonsei.ac.kr). Although a lot of time has passed, Can you provide me with a copy of the preprocess script of nuscenes dataset? Thank you very much.

**Comment by yanconglin** [comment]
Hi, could you share the nuscenes/Waymo preprocess script in the repo for easy access?  thank you :)

**Comment by matthew-wave** [comment]
Hello, what an amazing job.!
Please send a script and your nuscenes data structure diagram to my email 982007378@qq.com.

**Comment by matthew-wave** [comment]
> > [@PSYZ1234](https://github.com/PSYZ1234), Hello, I'm facing some challenges with the nuScenes dataset as well. If it's not too much trouble, could you share your data processing scripts?
> 
> Sure, please provide your email, and I will send it to you within the next few days.

Hello, what an amazing job!
Please send a script and your nuscenes data structure diagram to my email 982007378@qq.com.

**Comment by yanconglin** [comment]
Hi, any update on this issue? I am also looking for the training scripts for nuScenes. It would be great if you could open-source the scripts for the community. If not, could you please send a copy to my email (yanconglin0@gmail.com)—many thanks

**Comment by weiyixue999** [comment]
@yanconglin You can refer to the GeoNLF for Nuscenes-mini
